src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -110,24 +331,40 @@ private void AddDashboardResource(DistributedApplicationModel model)
         var fullyQualifiedDashboardPath = Path.GetFullPath(dashboardPath);
         var dashboardWorkingDirectory = Path.GetDirectoryName(fullyQualifiedDashboardPath);
 
-        ExecutableResource? dashboardResource = default;
+        // Create custom runtime config with AppHost's framework versions
+        var customRuntimeConfigPath = CreateCustomRuntimeConfig(fullyQualifiedDashboardPath);
 
+        // Find the dashboard DLL path
+        string dashboardDll;
         if (string.Equals(".dll", Path.GetExtension(fullyQualifiedDashboardPath), StringComparison.OrdinalIgnoreCase))
         {
-            // The dashboard path is a DLL, so run it with `dotnet <dll>`
-            dashboardResource = new ExecutableResource(KnownResourceNames.AspireDashboard, "dotnet", dashboardWorkingDirectory ?? "");
-
-            dashboardResource.Annotations.Add(new CommandLineArgsCallbackAnnotation(args =>
-            {
-                args.Add(fullyQualifiedDashboardPath);
-            }));
+            // Dashboard path is already a DLL
+            dashboardDll = fullyQualifiedDashboardPath;
         }
         else
         {
-            // Assume the dashboard path is directly executable
-            dashboardResource = new ExecutableResource(KnownResourceNames.AspireDashboard, fullyQualifiedDashboardPath, dashboardWorkingDirectory ?? "");
+            // Dashboard path is an executable, find the corresponding DLL
+            var directory = Path.GetDirectoryName(fullyQualifiedDashboardPath)!;
+            var baseName = Path.GetFileNameWithoutExtension(fullyQualifiedDashboardPath);
```

Comment:
This assumes the dashboard executable path is to a file with an extension, but for non-Windows platforms the executable file name will not have an extension, e.g. `Aspire.Dashboard`.